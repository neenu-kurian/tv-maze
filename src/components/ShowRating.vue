<template>
  <div class="flex gap-1 items-center text-white absolute">
    <StarIcon class="h-4 w-4 text-yellow-400 inline"/>
    <span>{{ score.toFixed(1) }}/10</span>
  </div>
</template>

<script setup lang="ts">
import { StarIcon } from '@heroicons/vue/24/solid';

defineProps<{
  score: number;
}>();
</script>

<style scoped>

</style>
