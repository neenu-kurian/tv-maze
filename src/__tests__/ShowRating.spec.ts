import { render } from '@testing-library/vue';
import ShowRating from '@/components/ShowRating.vue';
import { describe, it, expect } from 'vitest';

describe('ShowRating.vue', () => {
  it('matches snapshot', () => {
    const { container } = render(ShowRating, {
      props: {
        score: 8.5,
      },
    });
    expect(container).toMatchSnapshot();
  });
});
