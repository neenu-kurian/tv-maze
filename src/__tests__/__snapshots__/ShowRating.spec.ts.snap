// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeView.vue > matches snapshot 1`] = `
<div>
  <div
    class="flex gap-1 items-center text-white absolute"
  >
    <svg
      aria-hidden="true"
      class="h-4 w-4 text-yellow-400 inline"
      data-slot="icon"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M10.788 3.21c.448-1.077 1.976-1.077 2.424 0l2.082 5.006 5.404.434c1.164.093 1.636 1.545.749 2.305l-4.117 3.527 1.257 5.273c.271 1.136-.964 2.033-1.96 1.425L12 18.354 7.373 21.18c-.996.608-2.231-.29-1.96-1.425l1.257-5.273-4.117-3.527c-.887-.76-.415-2.212.749-2.305l5.404-.434 2.082-5.005Z"
        fill-rule="evenodd"
      />
    </svg>
    <span>
      8.5/10
    </span>
  </div>
</div>
`;

exports[`Show.vue > matches snapshot 1`] = `
<div>
  <div
    class="flex gap-1 items-center text-white absolute"
  >
    <svg
      aria-hidden="true"
      class="h-4 w-4 text-yellow-400 inline"
      data-slot="icon"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M10.788 3.21c.448-1.077 1.976-1.077 2.424 0l2.082 5.006 5.404.434c1.164.093 1.636 1.545.749 2.305l-4.117 3.527 1.257 5.273c.271 1.136-.964 2.033-1.96 1.425L12 18.354 7.373 21.18c-.996.608-2.231-.29-1.96-1.425l1.257-5.273-4.117-3.527c-.887-.76-.415-2.212.749-2.305l5.404-.434 2.082-5.005Z"
        fill-rule="evenodd"
      />
    </svg>
    <span>
      8.5/10
    </span>
  </div>
</div>
`;

exports[`ShowRating.vue > matches snapshot 1`] = `
<div>
  <div
    class="flex gap-1 items-center text-white absolute"
  >
    <svg
      aria-hidden="true"
      class="h-4 w-4 text-yellow-400 inline"
      data-slot="icon"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M10.788 3.21c.448-1.077 1.976-1.077 2.424 0l2.082 5.006 5.404.434c1.164.093 1.636 1.545.749 2.305l-4.117 3.527 1.257 5.273c.271 1.136-.964 2.033-1.96 1.425L12 18.354 7.373 21.18c-.996.608-2.231-.29-1.96-1.425l1.257-5.273-4.117-3.527c-.887-.76-.415-2.212.749-2.305l5.404-.434 2.082-5.005Z"
        fill-rule="evenodd"
      />
    </svg>
    <span>
      8.5/10
    </span>
  </div>
</div>
`;
